### ISSUE TITLE
Popup to get more user rating on app store

### ISSUE THREAD

**Issue by madhurgupta10** [commons-app/apps-android-commons#2508]
**Summary:** 
The app on The App Store needs more ratings and reviews, we can add a little popup displayed before the user closes the app which will ask the user if he/she wants to rate the app on app store. If user agrees then he/she will be redirected to The App Store or else user can opt for **DO NOT SHOW THIS AGAIN** option

**Screen-shots:** 
**_Example_**
![](http://ghostcode.in/wp-content/uploads/2016/11/rate-app-dialog@GhostCode.png)

**Would you like to work on the issue?**
Yes I want to work on this issue :)

**Comment by maskaravivek** [comment]
We might want to wait and be sure that all our issues around upload failures are fixed before we start prompting for ratings. Otherwise, we might attract more negative reviews than positive ones. :D

**Comment by madhurgupta10** [comment]
@maskaravivek Sure! I will be happy to add this feature as soon as issues around upload failures are fixed, can we keep the issue open?

**Comment by maskaravivek** [comment]
Yes, let us keep the issue open.

**Comment by nicolas-raoul** [comment]
We already have a link to rate the app in the About screen.
When the user quits the app, it is because they have something else to do (answer an email, look up directions, get the rare pokemon that just appeared, etc).
If we really really want to get more ratings, I would suggest showing the popup when the user opens their Achievements screen and they just got to the next level. The popup would say:

> Congratulations, you reached level 2!  
> The free encyclopedia appreciates your uploads.  
> Would you like to rate the upload app?  
> Yes - Ask me later - Don't ask me again

The user is more likely to be in a positive mood at that time.

Discussions about this topic:
- https://ux.stackexchange.com/questions/40717/asking-feedback-rating-for-a-mobile-app
- https://ux.stackexchange.com/questions/81625/best-practices-for-asking-user-to-rate-this-app
- https://ux.stackexchange.com/questions/57032/strategy-to-ask-app-users-for-review

**Comment by madhurgupta10** [comment]
@nicolas-raoul I agree that asking for feedback when the user reaches to next level would be much better than asking the user when he/she leaves the app.

**Comment by vanshikaarora** [comment]
I would like to pitch in my notions over the Issue.

>We already have a link to rate the app in the About screen.

Yes currently about has the playstore link to rate the app there might be very few users who will rate the app this way. 

>If we really really want to get more ratings, I would suggest showing the popup when the user opens their Achievements screen and they just got to the next level.

Yes definitely the user will be delighted to see the improvement in their score's and hence there is a greater chance of user's moving ahead and rating/giving positive reviews for the app.

Yet there are a few things to keep in mind:

- If the user hasn't clicked on "DO NOT SHOW THIS AGAIN" or if he has clicked on "REMIND ME LATER" then there user would be shown a popup every-time he navigates to Achievements
- Repeated popups are sometimes distracting and doesn't pleases the user
- The user won't be pleased to give good rating/reviews if this popup repeat's every-time he switches to achievements activity
- Considering the above point, we can display the popup only when the user has made a new achievement/ reached to the next level in achievements(For example if the user has reached a next level in achievements then while exiting the app we can show the popup _Congrats!! for reaching the next level. Please give sometime to rate our app on playstore_ )

**Comment by madhurgupta10** [comment]
@nicolas-raoul @maskaravivek is it a good time to implement this feature as per the discussion

> Congratulations, you reached level 2!
The free encyclopedia appreciates your uploads.
Would you like to rate the upload app?
Yes - Ask me later - Don't ask me again

**Comment by neslihanturan** [comment]
I think it is not yet, it requires 1, 2 months actually

**Comment by madhurgupta10** [comment]
@neslihanturan Okay

**Comment by Rishavgupta12345** [comment]
@nicolas-raoul @4D17Y4 i want to work on this issue please assign me

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345 Please work on one issue at a time, thanks for your enthusiasm! :-)

**Comment by Rishavgupta12345** [comment]
@4D17Y4 you can assign me this issue now

**Comment by 4D17Y4** [comment]
Hey @Rishavgupta12345,
Nice work so far. Now as you have some experience working on commons I'd say go for a difficult bug and try to solve it. 
Just to reiterate `A quality PR is more valuable than number of issues/PRs`.
 
With that said, finish up the assigned issues and then ping for this one. 
Thanks.

**Comment by Rishavgupta12345** [comment]
> Hey @Rishavgupta12345, Nice work so far. Now as you have some experience working on commons I'd say go for a difficult bug and try to solve it. Just to reiterate `A quality PR is more valuable than number of issues/PRs`.
> 
> With that said, finish up the assigned issues and then ping for this one. Thanks.

Thanks, @4D17Y4 I am targeting GSOC this year, it would be great to work under your guidance

**Comment by Rishavgupta12345** [comment]
@nicolas-raoul @4D17Y4 please assign me this

**Comment by Rishavgupta12345** [comment]
@nicolas-raoul @4D17Y4 worked on previous issues , so assign me this one

**Comment by nicolas-raoul** [comment]
This issue has the `debated` label, which means that it is not ready for a volunteer to work on.
The reason is still https://github.com/commons-app/apps-android-commons/issues/2508#issuecomment-466662817, upload is still failing very often.
Thanks! :-)
